modules/mapper-extras/src/yamlRestTest/resources/rest-api-spec/test/search-as-you-type/30_synthetic_source.yml

```diff
@@ -0,0 +1,48 @@
+setup:
+  - requires:
+      cluster_features: ["mapper.track_ignored_source"]
+      reason: requires tracking ignored source
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            _source:
+              mode: synthetic
+            properties:
+              a_field:
+                type: search_as_you_type
+
+---
+"synthetic source sanity test":
+  - do:
+      index:
+        index: test
+        id: "1"
+        body:
+          a_field: "quick brown fox jumps over a lazy dog"
+
+  - do:
+      index:
+        index: test
+        id: "2"
+        body:
+          a_field: null
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      get:
+        index: test
+        id: "1"
+
+  - match: { _source.a_field: "quick brown fox jumps over a lazy dog" }
+
+  - do:
+      get:
+        index: test
+        id: "2"
+
+  - match: { _source.a_field: null }
```

Comment:
Add an array test too?

Reply:
Great catch! I for some reason assumed that this field does not support arrays.